lib/synapse/retry_policy.rb

```diff
@@ -0,0 +1,45 @@
+module Synapse
+  module RetryPolicy
+    def with_retry(options = {}, &callback)
+      max_attempts = options['max_attempts'] || 1
+      max_delay = options['max_delay'] || Float::INFINITY
+      base_interval = options['base_interval'] || 0
```

Comment:
what about `first_interval`?

Reply:
what is the unit? I assume it is `millisecond`? might put some notes in the readme.

Reply:
it is seconds, let me document in the readme to make it obvious.